src/pages/openSourceApply.js

```diff
@@ -0,0 +1,170 @@
+      if (repositories.length !== 0) {
+        this.setState({ repositories, loadingRepos: false, result: {} });
+      } else {
+        this.setState({
+          loadingRepos: false,
+          result: { type: 'info', mesg: 'Info: No Repository found' },
+        });
+      }
+    } catch (error) {
+      this.setState({
+        loadingRepos: false,
+        result: { type: 'error', mesg: 'Error: An unknown error occured' },
+      });
+    }
+  }
+
+  renderContent() {
+    const { token, LoggedInUser } = this.props;
+    const { repositories } = this.state;
+
+    if (!LoggedInUser) {
+      return <SignInOrJoinFree redirect={Router.asPath} />;
+    } else if (!token || repositories.length === 0) {
+      return this.renderConnectGithubButton();
+    } else {
+      return this.renderGithubRepos();
+    }
+  }
+
+  async createCollectives(collectiveInputType) {
+    collectiveInputType.type = 'COLLECTIVE';
+    try {
+      const res = await this.props.createCollectiveFromGithub(collectiveInputType);
+      const collective = res.data.createCollectiveFromGithub;
+      Router.pushRoute('collective', {
+        slug: collective.slug,
+        status: 'collectiveCreated',
+        CollectiveId: collective.id,
+        CollectiveSlug: collective.slug,
+      });
+    } catch (err) {
+      console.error('>>> createCollective error: ', JSON.stringify(err)); // TODO - Remove
+      const errorMsg = err.graphQLErrors && err.graphQLErrors[0] ? err.graphQLErrors[0].message : err.message;
+      this.setState({
+        creatingCollective: false,
+        result: { type: 'error', mesg: errorMsg },
+      });
+      // throw new Error(errorMsg);
+    }
+  }
+
+  renderGithubRepos() {
+    const { repositories, creatingCollective } = this.state;
+    if (repositories.length !== 0) {
+      return (
+        <StyledInputField htmlFor="collective">
+          {fieldProps => (
+            <GithubRepositories
+              {...fieldProps}
+              repositories={repositories}
+              creatingCollective={creatingCollective}
+              onCreateCollective={data => {
+                this.setState({ creatingCollective: true });
+                this.createCollectives(data);
+              }}
+            />
+          )}
+        </StyledInputField>
+      );
+    }
+  }
+
+  renderConnectGithubButton() {
+    const connectUrl = `/api/connected-accounts/github?redirect=${WEBSITE_URL}/opensource/apply`;
+    return (
+      <StyledCard minWidth={400} maxWidth={500} border="none" minHeight={350} p={4} textAlign="center">
+        <H3 mb={2}>For Open Source projects</H3>
+        <P mb={4}>
+          You need a Github account and a repository with over 100 stars. If you run into trouble, file an issue in our
+          Github Issues section.
+        </P>
+        <StyledButton
+          textAlign="center"
+          buttonSize="large"
+          buttonStyle="primary"
+          onClick={() => {
```

Comment:
Can we make that a link instead?

Reply:
I used a link earlier but I wanted a way to disabled the link and show a loding spinner while the GitHub repos are being fetched.